R/pkg/R/DataFrame.R

```diff
@@ -2884,3 +2884,38 @@ setMethod("write.jdbc",
             write <- callJMethod(write, "mode", jmode)
             invisible(callJMethod(write, "jdbc", url, tableName, jprops))
           })
+
+#' randomSplit
+#'
+#' Return randomly split RDDs with the provided weights.
+#'
+#' @param x A SparkDataFrame
+#' @param weights a vector of weights for splits, will be normalized if they don't sum to 1
+#' @param seed Randomness seed value
+#'
+#' @family SparkDataFrame functions
+#' @rdname randomSplit
+#' @name randomSplit
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlContext <- sparkRSQL.init(sc)
+#' df <- createDataFrame(data.frame(id = 1:1000))
+#' df_list <- randomSplit(df, c(2, 3, 5), 0)
+#' sapply(df_list, count)
+#'}
+setMethod("randomSplit",
+          signature(x = "SparkDataFrame", weights = "numeric"),
+          function(x, weights, seed) {
+            if (!all(sapply(weights, function(c) { c >= 0 }))) {
+              stop("all weight values should not be negative")
+            }
+            normalized_list <- as.list(weights / sum(weights))
+            if (!missing(seed)) {
```

Comment:
since this is specified in the signature, it wouldn't be possible to omit seed - if that's what you want to support, you can take the weight from signature() but you will need to check/coerce into numeric to make sure